ui/partials/perf/testdatachooser.html

```diff
@@ -2,19 +2,47 @@
   <button type="button" class="close" ng-click="cancel()"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
   <h3 class="modal-title">Add test data</h3>
 </div>
+<div class="test-data-chooser">
 <div class="modal-body">
-  <h4>Project</h4>
-  <select class="form-control" ng-change="updateTestInput()" ng-model="selectedProject" ng-options="project.name for project in projects">
-  </select>
-  <h4>Platform</h4>
-  <p class="blink" ng-show="loadingTestData">Loading series data...</p>
-  <select ng-hide="loadingTestData" ng-change="updateTestSelector()" ng-model="selectedPlatform" ng-options="platform for platform in platformList">
-  </select>
-  <h4>Test</h4>
-  <p class="blink" ng-show="loadingTestData">Loading series data...</p>
-  <div id="scrollable-dropdown-menu">
-    <input my-input ng-hide="loadingTestData" class="typeahead" type="text" placeholder="Enter test keywords (e.g. 'tp5o')"></input>
+  <div class="test-chooser">
+    <h4>Project</h4>
+    <select class="form-control" ng-change="updateTestInput()" ng-model="selectedProject" ng-options="project.name for project in projects">
+    </select>
+    <h4>Platform</h4>
+    <p class="blink" ng-show="loadingTestData">Loading series data...</p>
+    <select ng-hide="loadingTestData" ng-change="updateTestSelector()" ng-model="selectedPlatform" ng-options="platform for platform in platformList">
+    </select>
+    <h4>Tests</h4>
+    <div class="form-group">
+        <input class="form-control input-sm" type="text" ng-model="test_filter" placeholder="Please enter string to select tests"/>
+    </div>
+    <div id="choose-test-list" class="form-group">
+      <select multiple class="form-control"  ng-model="leftSelected">
+         <option value="{{::left_elem}}" ng-repeat="left_elem in LeftList| filter: test_filter track by $id(left_elem)">
+            {{::left_elem}}
+         </option>     
+      </select>
+    </div>
   </div>
+  <div class="btn-group-vertical ">
+     <button ng-click="move_left()" type="button" class="btn btn-xs btn-default">
+        <span class="glyphicon glyphicon-chevron-left"></span>
+     </button>
+     <button ng-click="move_right()" type="button" class="btn btn-xs btn-default">
+        <span class="glyphicon glyphicon-chevron-right"></span>
+     </button>
+  </div>
+  <div class="test-list-containter">
+    <h4>Tests to add</h4>
+    <div id="added-test-list" class="form-group">
+      <select class="form-control" multiple ng-model="rightSelected">
+         <option value="{{ ::right_elem }}" ng-repeat="right_elem in rightList">
+           {{ ::right_elem }}
```

Comment:
I know the repo has a mix of angular expressions with spaces between the braces (both regular and one-way), but perhaps we'd want to match the styleguide used in the angular docs and your L22 farther up, which is no spaces. I defer to Will though.
https://docs.angularjs.org/guide/expression